xbmc/LangInfo.cpp

```diff
@@ -626,31 +645,25 @@ const std::string& CLangInfo::GetSpeedUnitString() const
   return g_localizeStrings.Get(SPEED_UNIT_STRINGS+m_currentRegion->m_speedUnit);
 }
 
-void CLangInfo::SettingOptionsLanguagesFiller(const CSetting *setting, std::vector< std::pair<std::string, std::string> > &list, std::string &current, void *data)
+std::set<std::string> CLangInfo::GetSortTokens() const
 {
-  //find languages...
-  CFileItemList items;
-  XFILE::CDirectory::GetDirectory("special://xbmc/language/", items);
+  std::set<std::string> sortTokens = m_sortTokens;
+  sortTokens.insert(g_advancedSettings.m_vecTokens.begin(), g_advancedSettings.m_vecTokens.end());
 
-  vector<string> vecLanguage;
-  for (int i = 0; i < items.Size(); ++i)
-  {
-    CFileItemPtr pItem = items[i];
-    if (pItem->m_bIsFolder)
-    {
-      if (StringUtils::EqualsNoCase(pItem->GetLabel(), ".svn") ||
-          StringUtils::EqualsNoCase(pItem->GetLabel(), "fonts") ||
-          StringUtils::EqualsNoCase(pItem->GetLabel(), "media"))
-        continue;
+  return sortTokens;
+}
 
-      vecLanguage.push_back(pItem->GetLabel());
-    }
-  }
+void CLangInfo::SettingOptionsLanguagesFiller(const CSetting *setting, std::vector< std::pair<std::string, std::string> > &list, std::string &current, void *data)
+{
+  // find languages...
+  ADDON::VECADDONS addons;
+  if (!ADDON::CAddonMgr::Get().GetAddons(ADDON::ADDON_RESOURCE_LANGUAGE, addons, true))
+    return;
 
-  sort(vecLanguage.begin(), vecLanguage.end(), sortstringbyname());
+  for (ADDON::VECADDONS::const_iterator addon = addons.begin(); addon != addons.end(); ++addon)
+    list.push_back(make_pair((*addon)->Name(), boost::shared_dynamic_cast<ADDON::CLanguageResource>(*addon)->GetLocaleLC()));
```

Comment:
Doesn't compile for me. boost::shared_dynamic_cast was deprecated and removed in 1.53 apparently